syntax = "proto3";

package silenteight.agent.gender.v1.api;

option java_package = "com.silenteight.proto.agent.gender.v1.api";
option java_outer_classname = "GenderAgentApiProto";
option java_multiple_files = true;

service GenderAgent {
    rpc SolveGender (SolveGenderRequest) returns (stream SolveGenderResponse) {
    }
}

message SolveGenderRequest {
  string alerted_value = 1 [deprecated = true];
  repeated string matched_values = 2;
  repeated string alerted_values = 3;
}

message SolveGenderResponse {
  string result = 1;
  SolveGenderReason reason = 2;
}

message SolveGenderReason {
    string customer_value = 1 [deprecated = true];
    repeated string watchlist_values = 2;
    repeated string customer_values = 3;
}
